keep/api/tasks/process_topology_task.py

```diff
@@ -36,10 +36,13 @@ def process_topology(
             "Deleting existing topology data",
             extra=extra,
         )
-        session.query(TopologyService).filter(
+        services = session.query(TopologyService).filter(
             TopologyService.source_provider_id == provider_id,
             TopologyService.tenant_id == tenant_id,
-        ).delete()
+        ).all()
+
+        for service in services:
+            session.delete(service)
```

Comment:
This is very bad since it runs a query for every service. Why not delete all at once?